R/pr.R

```diff
@@ -161,34 +197,46 @@ pr_push <- function() {
 
 #' @export
 #' @rdname pr_init
-pr_pull <- function() {
+pr_pull <- function(credentials = NULL,
+                    auth_token = github_token()) {
   check_uses_github()
   check_branch_not_master()
   check_uncommitted_changes()
 
+  protocol <- github_remote_protocol()
+  credentials <- credentials %||% git2r_credentials(protocol, auth_token)
+
   ui_done("Pulling changes from GitHub PR")
-  git_pull()
+  git_pull(credentials = credentials)
 
   invisible(TRUE)
 }
 
 #' @export
 #' @rdname pr_init
-pr_pull_upstream <- function() {
+pr_pull_upstream <- function(credentials = NULL,
+                             auth_token = github_token()) {
   check_uses_github()
   check_uncommitted_changes()
 
+  protocol <- github_remote_protocol()
+  credentials <- credentials %||% git2r_credentials(protocol, auth_token)
+
   source <- if (git_is_fork()) "upstream/master" else "origin/master"
+  ## TODO: this claim seems a bit like wishful thinking ... what assumptions
+  ## are we making about the setup here?
   ui_done("Pulling changes from GitHub source repo {ui_value(source)}")
-  git_pull(source)
+  ## TODO: why are we not passing `source`?
```

Comment:
Why aren't you?

Reply:
OK I reworked this so that we are more clearly doing what we say we are doing. This function strikes me as something people might surprise themselves with (i.e. pulling master into whatever branch they are on). But I guess that is its point.